<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>app</class>
 <widget class="QMainWindow" name="app">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>664</width>
    <height>621</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Flasher GUI</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>621</width>
      <height>211</height>
     </rect>
    </property>
    <property name="tabPosition">
     <enum>QTabWidget::North</enum>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>avrdude</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>591</width>
        <height>61</height>
       </rect>
      </property>
      <property name="title">
       <string>Core</string>
      </property>
      <widget class="QRadioButton" name="optM8">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>100</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>atmega8</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QRadioButton" name="optM16">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>30</y>
         <width>100</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>atmega16</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="optM32">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>30</y>
         <width>100</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>atmega32</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="optM8535">
       <property name="geometry">
        <rect>
         <x>270</x>
         <y>30</y>
         <width>100</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>atmega8535</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="optM1280">
       <property name="geometry">
        <rect>
         <x>370</x>
         <y>30</y>
         <width>100</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>atmega1280</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="optM2560">
       <property name="geometry">
        <rect>
         <x>470</x>
         <y>30</y>
         <width>100</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>atmega2560</string>
       </property>
      </widget>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>57</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>Main App</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtAvrApp">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>80</y>
        <width>113</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>avrdude</string>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>110</y>
        <width>57</width>
        <height>15</height>
       </rect>
      </property>
      <property name="cursor">
       <cursorShape>ArrowCursor</cursorShape>
      </property>
      <property name="text">
       <string>Tool</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtAvrTool">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>110</y>
        <width>113</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>usbasp</string>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>140</y>
        <width>57</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>Port</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtAvrPort">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>140</y>
        <width>113</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>usb</string>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>80</y>
        <width>271</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>File Path (Format Intel Hexadecimal *.hex)</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtAvrFile">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>100</y>
        <width>331</width>
        <height>23</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="btnAvrBrowse">
      <property name="geometry">
       <rect>
        <x>550</x>
        <y>100</y>
        <width>41</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Find</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnAvrWrite">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>130</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Write</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnAvrErase">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>130</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Erase</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnAvrTest">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>130</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Test</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>st-link</string>
     </attribute>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>10</y>
        <width>271</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>File Path (Format File RAW Binary *.bin)</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnStlinkErase">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>60</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Erase</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnStlinkBrowse">
      <property name="geometry">
       <rect>
        <x>550</x>
        <y>30</y>
        <width>41</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Find</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnStlinkWrite">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>60</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Write</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtStlinkFile">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>30</y>
        <width>331</width>
        <height>23</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>57</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>Main App</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtStlinkApp">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>10</y>
        <width>113</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>st-flash</string>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtStlinkTool">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>40</y>
        <width>113</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>st-link</string>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>57</width>
        <height>15</height>
       </rect>
      </property>
      <property name="cursor">
       <cursorShape>ArrowCursor</cursorShape>
      </property>
      <property name="text">
       <string>Tool</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtStlinkPort">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>70</y>
        <width>113</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>usb</string>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>57</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>Port</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>stm32flash</string>
     </attribute>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>57</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>Port</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>57</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>Main App</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnSTFlashWrite">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>60</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Write</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtSTFlashTool">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>40</y>
        <width>113</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>serial-ftdi</string>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="btnSTFlashBrowse">
      <property name="geometry">
       <rect>
        <x>550</x>
        <y>30</y>
        <width>41</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Find</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>57</width>
        <height>15</height>
       </rect>
      </property>
      <property name="cursor">
       <cursorShape>ArrowCursor</cursorShape>
      </property>
      <property name="text">
       <string>Tool</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtSTFlashApp">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>10</y>
        <width>113</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>stm32flash</string>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="btnSTFlashTest">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>60</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Test</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtSTFlashPort">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>70</y>
        <width>113</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>usb</string>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtSTFlashFile">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>30</y>
        <width>331</width>
        <height>23</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="btnSTFlashErase">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>60</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Erase</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>10</y>
        <width>271</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>File Path (Format File RAW Binary *.bin)</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>100</y>
        <width>57</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>Serial</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtSTFlashSerial">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>100</y>
        <width>113</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>/dev/ttyUSB0</string>
      </property>
      <property name="readOnly">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtSTFlashBaudrate">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>130</y>
        <width>113</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>default</string>
      </property>
      <property name="readOnly">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>57</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>Baudrate</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QTextEdit" name="txtOutput">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>230</y>
      <width>621</width>
      <height>301</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>7</pointsize>
     </font>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>664</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionQuit"/>
   <addaction name="actionHelp"/>
   <addaction name="actionAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionHelp">
   <property name="text">
    <string>Help</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
